### ISSUE TITLE
Support `gpt-4-vision-preview`

### ISSUE THREAD

**Issue by PaarthShah** [matrixgpt/matrix-chatgpt-bot#247]
https://platform.openai.com/docs/guides/vision

It seems like [uploading base64-encoded images](https://platform.openai.com/docs/guides/vision/uploading-base-64-encoded-images) may be a generic viable strategy for passing images to the AI.

Alternatively/for speed, and from unencrypted rooms, it may instead be possible/desirable to pass an image URL by transforming the image `mxc` url to an `https` url via the `image_url` key.

**Comment by max298** [comment]
As far as I can tell we're limited by the [library we use](https://github.com/waylaidwanderer/node-chatgpt-api) for API communication, which does not yet support vision. Although I'm very interested and will check what we can do as soon as the library adds support.

**Comment by Dual-0** [comment]
I open up [a request](https://github.com/waylaidwanderer/node-chatgpt-api/issues/504).

**Comment by max298** [comment]
I think we might consider dropping the third party SDK and switch to the official node package from openai: https://github.com/openai/openai-node#readme which seems to [support vision](https://github.com/openai/openai-openapi/pull/107/files)

**Comment by PaarthShah** [comment]
Going for the official node library seems like the best option for long-term sustainability and rapid adoption of new features

**Comment by bertybuttface** [comment]
Yes but we would then be responsible for handling context, which is fine if someone is willing to write the code.
